Sources/CoreFoundation/CFKnownLocations.c

```diff
@@ -96,30 +97,16 @@ CFURLRef _Nullable _CFKnownLocationCreatePreferencesURLForUser(CFKnownLocationUs
         }
         case _kCFKnownLocationUserCurrent:
             username = CFGetUserName();
-            // fallthrough
+            break;
```

Comment:
This seems wrong. It is backwards no? `LocalAppData` would be for the current user, not a specified user. This seems like it would not return a location for the current user, and return the location for the current user when you specify a user name.

Reply:
Quoting @stevenbrix for his attention.

Reply:
@compnerd I wonder whether the entire case `case _kCFKnownLocationUserCurrent` can be written as the following instead:
```c
        case _kCFKnownLocationUserCurrent:
            username = CFGetUserName();
            if !(CFEqual(username, CFSTR(""))) {
                break;
            }
            // fallthrough.
```